skynet/dgpu/network.py

```diff
@@ -177,49 +156,86 @@ async def _run_and_save(d, key: str, fn, *args, **kwargs):
             d[key] = await fn(*args, **kwargs)
 
         async with trio.open_nursery() as n:
-            n.start_soon(_run_and_save, snap, 'my_results', self.find_my_results)
+            n.start_soon(_run_and_save, snap, 'results', self.find_results)
             for req in snap['queue']:
                 n.start_soon(
                     _run_and_save, snap['requests'], req['id'], self.get_status_by_request_id, req['id'])
 
+
+        maybe_update_tui(lambda tui: tui.network_update(snap))
+
         return snap
 
+    async def wait_data_update(self):
+        await self._data_event.wait()
+
+    async def iter_poll_update(self, poll_time: float):
```

Comment:
much better and preps us for an underlying streaming IPC from the smart-con in the future 😎